File: LeetCode/iter1/c++/693.binary-number-with-alternating-bits.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <vector>
#include <sstream>
#include <unordered_map>
#include <unordered_set>
#include <utility>
#include <queue>
#include <stack>

using namespace std;

class Solution {
public:
	bool hasAlternatingBits(int n) {
		bool ret = true;
		int digit = n & 1, oldDigit;
		while (n != 0) {
			n = n >> 1;
			oldDigit = n & 1;
			if (digit == oldDigit) {
				ret = false;
				break;
			}
			digit = oldDigit;
		}

		return ret;
	}
};

int main()
{
	Solution solution;
	for (int i = 0; i < 30; i++) {
		cout << i << " " << (solution.hasAlternatingBits(i) ? "Yes\n" : "No\n");
	}

	return 0;
}

File: LeetCode/iter1/java/693.binary-number-with-alternating-bits.java
package org.ironman.base.playground.leetcode;

public class Solution {

    public static void main(String[] args) {
        for (int i = 0; i < 30; i++) {
            System.out.println(i + " " + hasAlternatingBits(i));
        }
    }

    public static boolean hasAlternatingBits(int n) {
        boolean ret = true;
        int curDigit = n & 1, preDigit = curDigit;

        while (n != 0) {
            n = n >> 1;
            curDigit = n & 1;
            if (curDigit == preDigit) {
                ret = false;
                break;
            }

            preDigit = curDigit;
        }

        return ret;
    }

}